test/Common/MoistThermodynamics/runtests.jl

```diff
@@ -237,6 +237,9 @@ end
 
     # Accurate but expensive `LiquidIcePotTempSHumNonEquil` constructor (Non-linear temperature from θ_liq_ice)
     T_non_linear = air_temperature_from_liquid_ice_pottemp_non_linear.(θ_liq_ice, ρ, FT(1e-3), 10, q_pt)
+    T_expansion = air_temperature_from_liquid_ice_pottemp.(θ_liq_ice, ρ, q_pt)
+    C_1 = 0.0001
+    @test all(abs.(T_non_linear .- T_expansion) .< 15*sqrt(eps(FT)) .+ C_1 .* latent_heat_liq_ice.(q_pt)./cv_m.(q_pt))
```

Comment:
Simpler: I'd test for a given relative error (`rtol` a few percent) between `T_non_linear` and `T_expansion`.

Reply:
Ah, good point. Adjusted, works, and looks even better.